### PR TITLE
Breakpoints Issue - clear breakpoints when close all notebooks

### LINKED ISSUE #168: Breakpoints not removed from the sidebar on notebook closed
When a notebook is closed, the breakpoints stay in the sidebar and are not removed.

Clicking on "Remove All Breakpoints" throws the following error:

```
Uncaught (in promise) Error: A kernel is required to send debug messages.
```

breakpoints-notebook-closed

### PR DESCRIPTION
**Pull request by KsavinN**
resolve #168

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -103,6 +104,10 @@ class DebuggerHandler<H extends ConsoleHandler | NotebookHandler> {
       widget.disposed.connect(() => {
         handler.dispose();
         delete this.handlers[widget.id];
+        if (Object.keys(this.handlers).length === 0) {
```
[the remark this answers is not in the document]
Comment (KsavinN) [reply src/index.ts]: I implement that in that place cause it works now for notebook and consoles.

Ok. Should we clear debugging information like breakpoints only or just all?

**Comment by KsavinN** [comment]
I check if we can just stop session for not supported widget and restart but its seems like after that going to back the notebook not restore breakpoints

**Comment by jtpio** [comment]
Do the logs in xeus-python or the dev tools say anything?

**Comment by KsavinN** [comment]
> Do the logs in xeus-python or the dev tools say anything?

stop_debug

**Comment by KsavinN** [comment]
> > Do the logs in xeus-python or the dev tools say anything?

actually this is good behaviour. The stop request needs to send before the widget is disposed or kernel changed.

**Comment by jtpio** [comment]
Which one is good behavior?

**Comment by KsavinN** [comment]
> Which one is good behaviour?

The method Stop debugger doesn't work due to no kernel. After close widget kernel is disposed. That's why that don't work.

**Comment by KsavinN** [comment]
I tried to resolve additionally this issue #215. But i guess, it will take more time for me or i will need some suggestions. For now, it only resolves #168

**Comment by KsavinN** [comment]
Close due to That PR source belongs to not existing fork.

**Pull request closed without merging**